SierraSoftworks/tailscale-udm
# feat: Add initial logic for UDM-SE support

This pull request includes the first pass of adding support for the UDR/UDM-SE (and other UniFi OS 2.x devices) to this repository.

It does so by expanding the `manage.sh` script to support a plugin-style architecture which will automatically load the correct UniFi OS plugin module (1.x or 2.x) which then provides custom integration logic for each of these operating systems. UniFi OS 2.x (which is installed on the UDR and UDM-SE) is a ground-up change of how the UniFi operating system works, and removes the need for the boostchicken/on_boot script used to ensure persistence on the UDM/UDM-Pro.

Instead, we rely on the standard Tailscale `.deb` package, installing it using `dpkg -i` and then configuring `/etc/default/tailscaled` to turn on userspace networking. You'll find this implementation (and the corresponding wrappers around `systemctl start/stop/restart`) in the `package/unios_2.x.sh` file - which is the plugin module for UniFi OS 2.x.

:warning: While this PR has a test suite associated with it, it has **NOT** been tested on a live UDR/UDM-SE yet, your mileage may vary. Feedback and testing would be greatly appreciated, but please do not do so on a production device in case anything goes wrong.

## Testing Steps
To validate that this new release works correctly on your device, please follow these steps to install a prerelease version of it:

1. Download the latest release package (from the `Build / Package` pipeline) onto your device.
   ```sh
   curl -sSLq --ipv4 -o /tmp/tailscale-udm-pre.zip https://github.com/SierraSoftworks/tailscale-udm/suites/9812203326/artifacts/472244022
   ```
2. Unzip the build artifact package using `unzip -oq /tmp/tailscale-udm-pre.zip -d /tmp`
3. Un-tar the installation package into your `/data/tailscale` directory using `tar xzf /tmp/tailscale-udm.tar.gz -C /data/`
4. Run the installation command: `/data/tailscale/manage.sh install`
5. Start tailscale using `/data/tailscale/manage.sh start`

Report if you encounter any issues with the above and/or installation does not complete as expected.
